--- src/tModLoader/Terraria/GameContent/UI/EmoteBubble.cs
+++ src/tStandalone/Terraria/GameContent/UI/EmoteBubble.cs
@@ -9,7 +_,7 @@
 {
 	public class EmoteBubble
 	{
-		internal static int[] CountNPCs = new int[NPCID.Count];
+		private static int[] CountNPCs = new int[NPCID.Count];
 		public static Dictionary<int, EmoteBubble> byID = new Dictionary<int, EmoteBubble>();
 		private static List<int> toClean = new List<int>();
 		public static int NextID;
@@ -337,7 +_,7 @@
 		}
 
 		private void ProbeTownNPCs(List<int> list) {
-			for (int i = 0; i < 663; i++) {
+			for (int i = 0; i < NPCID.Count; i++) {
 				CountNPCs[i] = 0;
 			}
 
@@ -347,7 +_,7 @@
 			}
 
 			int type = ((NPC)anchor.entity).type;
-			for (int k = 0; k < 663; k++) {
+			for (int k = 0; k < NPCID.Count; k++) {
 				if (NPCID.Sets.FaceEmote[k] > 0 && CountNPCs[k] > 0 && k != type)
 					list.Add(NPCID.Sets.FaceEmote[k]);
 			}
