--- src/Terraria/Terraria/ID/ProjectileID.cs
+++ src/tModLoader/Terraria/ID/ProjectileID.cs
@@ -6,7 +_,7 @@
 	{
 		public static class Sets
 		{
-			public static SetFactory Factory = new SetFactory(950);
+			public static SetFactory Factory = new SetFactory(Count);
 			public static bool[] DontApplyParryDamageBuff = Factory.CreateBoolSet(false, 524, 321, 181, 566);
 			public static bool[] IsAWhip = Factory.CreateBoolSet(false, 847, 841, 848, 849, 912, 913, 914, 915);
 			public static bool[] ImmediatelyUpdatesNPCBuffFlags = Factory.CreateBoolSet(636);
@@ -1209,6 +_,7 @@
 		public const short Mace = 947;
 		public const short FlamingMace = 948;
 		public const short TorchGod = 949;
-		public const short Count = 950;
+		public const short KingSlimeShockwave = 950;
+		public const short Count = 951;
 	}
 }
