--- src/Terraria/Terraria/IO/FileData.cs
+++ src/tModLoader/Terraria/IO/FileData.cs
@@ -10,6 +_,7 @@
 		public string Name;
 		public readonly string Type;
 		protected bool _isFavorite;
+		protected bool _isDisplayingModData;
 
 		public string Path => _path;
 
@@ -17,6 +_,8 @@
 
 		public bool IsFavorite => _isFavorite;
 
+		public bool IsDisplayingModData => _isDisplayingModData;
+
 		protected FileData(string type) {
 			Type = type;
 		}
@@ -26,18 +_,30 @@
 			_path = path;
 			_isCloudSave = isCloud;
 			_isFavorite = (isCloud ? Main.CloudFavoritesData : Main.LocalFavoriteData).IsFavorite(this);
+			_isDisplayingModData = Main.LocalModInfoData.IsDisplayingModData(this);
 		}
 
 		public void ToggleFavorite() {
 			SetFavorite(!IsFavorite);
 		}
 
+		public void ToggleModData() {
+			SetModData(!IsDisplayingModData);
+		}
+
 		public string GetFileName(bool includeExtension = true) => FileUtilities.GetFileName(Path, includeExtension);
 
 		public void SetFavorite(bool favorite, bool saveChanges = true) {
 			_isFavorite = favorite;
 			if (saveChanges)
 				(IsCloudSave ? Main.CloudFavoritesData : Main.LocalFavoriteData).SaveFavorite(this);
+		}
+
+		public void SetModData(bool modData, bool saveChanges = true) {
+			_isDisplayingModData = modData;
+			if (saveChanges) {
+				Main.LocalModInfoData.SaveModInfo(this);
+			}
 		}
 
 		public abstract void SetAsActive();
