--- src/tModLoader/Terraria/GameContent/ItemDropRules/ItemDropDatabase.cs
+++ src/tStandalone/Terraria/GameContent/ItemDropRules/ItemDropDatabase.cs
@@ -394,6 +_,9 @@
 			leadingConditionRule2.OnSuccess(ItemDropRule.Common(1225, 1, 15, 30));
 			leadingConditionRule2.OnSuccess(ItemDropRule.Common(549, 1, 25, 40));
 			leadingConditionRule.OnSuccess(ItemDropRule.MasterModeCommonDrop(4931));
+
+			leadingConditionRule.OnSuccess(ItemDropRule.MasterModeCommonDrop(ItemID.TwinsGun));
+
 			leadingConditionRule.OnSuccess(ItemDropRule.MasterModeDropOnAllPlayers(4804, _masterModeDropRng));
 			RegisterToMultipleNPCs(leadingConditionRule, 126, 125);
 		}
