--- src/Terraria/Terraria/DataStructures/PlayerDrawSet.cs
+++ src/tStandalone/Terraria/DataStructures/PlayerDrawSet.cs
@@ -479,17 +_,17 @@
 			bodyGlowColor = drawPlayer.GetImmuneAlphaPure(bodyGlowColor, alphaReduction);
 			armGlowColor = drawPlayer.GetImmuneAlphaPure(armGlowColor, alphaReduction);
 			legsGlowColor = drawPlayer.GetImmuneAlphaPure(legsGlowColor, alphaReduction);
-			if (drawPlayer.head > 0 && drawPlayer.head < 266) {
+			if (drawPlayer.head > 0 && drawPlayer.head < ArmorIDs.Head.Count) {
 				Main.instance.LoadArmorHead(drawPlayer.head);
 				int num6 = ArmorIDs.Head.Sets.FrontToBackID[drawPlayer.head];
 				if (num6 >= 0)
 					Main.instance.LoadArmorHead(num6);
 			}
 
-			if (drawPlayer.body > 0 && drawPlayer.body < 235)
+			if (drawPlayer.body > 0 && drawPlayer.body < ArmorIDs.Body.Count)
 				Main.instance.LoadArmorBody(drawPlayer.body);
 
-			if (drawPlayer.legs > 0 && drawPlayer.legs < 218)
+			if (drawPlayer.legs > 0 && drawPlayer.legs < ArmorIDs.Legs.Count)
 				Main.instance.LoadArmorLegs(drawPlayer.legs);
 
 			if (drawPlayer.handon > 0 && drawPlayer.handon < 22)
@@ -1186,7 +_,7 @@
 		private void CreateCompositeData() {
 			frontShoulderOffset = Vector2.Zero;
 			backShoulderOffset = Vector2.Zero;
-			usesCompositeTorso = (drawPlayer.body > 0 && drawPlayer.body < 235 && ArmorIDs.Body.Sets.UsesNewFramingCode[drawPlayer.body]);
+			usesCompositeTorso = (drawPlayer.body > 0 && drawPlayer.body < ArmorIDs.Body.Count && ArmorIDs.Body.Sets.UsesNewFramingCode[drawPlayer.body]);
 			usesCompositeFrontHandAcc = (drawPlayer.handon > 0 && drawPlayer.handon < 22 && ArmorIDs.HandOn.Sets.UsesNewFramingCode[drawPlayer.handon]);
 			usesCompositeBackHandAcc = (drawPlayer.handoff > 0 && drawPlayer.handoff < 14 && ArmorIDs.HandOff.Sets.UsesNewFramingCode[drawPlayer.handoff]);
 			if (drawPlayer.body < 1)
