--- src/tModLoader/Terraria/DataStructures/PlayerDrawLayers.cs
+++ src/tStandalone/Terraria/DataStructures/PlayerDrawLayers.cs
@@ -1766,7 +_,7 @@
 		}
 
 		public static void DrawPlayer_25_Shield(ref PlayerDrawSet drawinfo) {
-			if (drawinfo.drawPlayer.shield <= 0 || drawinfo.drawPlayer.shield >= TextureAssets.AccShield.Length)
+			if (drawinfo.drawPlayer.shield <= 0 || drawinfo.drawPlayer.shield >= 10)
 				return;
 
 			Vector2 zero = Vector2.Zero;
