--- src/Terraria/Terraria/DataStructures/PlayerDrawHeadLayers.cs
+++ src/tStandalone/Terraria/DataStructures/PlayerDrawHeadLayers.cs
@@ -14,7 +_,7 @@
 		}
 
 		public static void DrawPlayer_00_BackHelmet(ref PlayerDrawHeadSet drawinfo) {
-			if (drawinfo.drawPlayer.head >= 0 && drawinfo.drawPlayer.head < 266) {
+			if (drawinfo.drawPlayer.head >= 0 && drawinfo.drawPlayer.head < ArmorIDs.Head.Count) {
 				int num = ArmorIDs.Head.Sets.FrontToBackID[drawinfo.drawPlayer.head];
 				if (num >= 0) {
 					Rectangle hairFrame = drawinfo.HairFrame;
