--- src/tModLoader/Terraria/DataStructures/PlayerDrawLayers.cs
+++ src/tStandalone/Terraria/DataStructures/PlayerDrawLayers.cs
@@ -223,7 +_,7 @@
 		}
 
 		public static void DrawPlayer_01_3_BackHead(ref PlayerDrawSet drawinfo) {
-			if (drawinfo.drawPlayer.head >= 0 && drawinfo.drawPlayer.head < 266) {
+			if (drawinfo.drawPlayer.head >= 0 && drawinfo.drawPlayer.head < ArmorIDs.Head.Count) {
 				int num = ArmorIDs.Head.Sets.FrontToBackID[drawinfo.drawPlayer.head];
 				if (num >= 0) {
 					Vector2 helmetOffset = drawinfo.helmetOffset;
@@ -886,7 +_,7 @@
 			float rotation = bodyRotation + drawinfo.compositeBackArmRotation;
 			bool flag = !drawinfo.drawPlayer.invis;
 			bool flag2 = !drawinfo.drawPlayer.invis;
-			bool flag3 = drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < 235;
+			bool flag3 = drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < ArmorIDs.Body.Count;
 			bool flag4 = !drawinfo.hidesTopSkin;
 			bool flag5 = false;
 			DrawData drawData;
@@ -983,7 +_,7 @@
 		public static void DrawPlayer_13_Leggings(ref PlayerDrawSet drawinfo) {
 			DrawData item;
 			if (drawinfo.isSitting && drawinfo.drawPlayer.legs != 140 && drawinfo.drawPlayer.legs != 217) {
-				if (drawinfo.drawPlayer.legs > 0 && drawinfo.drawPlayer.legs < 218 && (!ShouldOverrideLegs_CheckShoes(ref drawinfo) || drawinfo.drawPlayer.wearsRobe)) {
+				if (drawinfo.drawPlayer.legs > 0 && drawinfo.drawPlayer.legs < ArmorIDs.Legs.Count && (!ShouldOverrideLegs_CheckShoes(ref drawinfo) || drawinfo.drawPlayer.wearsRobe)) {
 					if (!drawinfo.drawPlayer.invis) {
 						DrawSittingLegs(ref drawinfo, TextureAssets.ArmorLeg[drawinfo.drawPlayer.legs].Value, drawinfo.colorArmorLegs, drawinfo.cLegs);
 						if (drawinfo.legsGlowMask != -1)
@@ -1021,7 +_,7 @@
 					drawinfo.DrawDataCache.Add(item);
 				}
 			}
-			else if (drawinfo.drawPlayer.legs > 0 && drawinfo.drawPlayer.legs < 218 && (!ShouldOverrideLegs_CheckShoes(ref drawinfo) || drawinfo.drawPlayer.wearsRobe)) {
+			else if (drawinfo.drawPlayer.legs > 0 && drawinfo.drawPlayer.legs < ArmorIDs.Legs.Count && (!ShouldOverrideLegs_CheckShoes(ref drawinfo) || drawinfo.drawPlayer.wearsRobe)) {
 				if (drawinfo.drawPlayer.invis)
 					return;
 
@@ -1123,7 +_,7 @@
 		}
 
 		public static void DrawPlayer_15_SkinLongCoat(ref PlayerDrawSet drawinfo) {
-			if ((drawinfo.skinVar == 3 || drawinfo.skinVar == 8 || drawinfo.skinVar == 7) && (drawinfo.drawPlayer.body <= 0 || drawinfo.drawPlayer.body >= 235) && !drawinfo.drawPlayer.invis) {
+			if ((drawinfo.skinVar == 3 || drawinfo.skinVar == 8 || drawinfo.skinVar == 7) && (drawinfo.drawPlayer.body <= 0 || drawinfo.drawPlayer.body >= ArmorIDs.Body.Count) && !drawinfo.drawPlayer.invis) {
 				if (drawinfo.isSitting) {
 					DrawSittingLegs(ref drawinfo, TextureAssets.Players[drawinfo.skinVar, 14].Value, drawinfo.colorShirt);
 					return;
@@ -1212,7 +_,7 @@
 			if (drawinfo.usesCompositeTorso) {
 				DrawPlayer_17_TorsoComposite(ref drawinfo);
 			}
-			else if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < 235) {
+			else if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < ArmorIDs.Body.Count) {
 				Rectangle bodyFrame = drawinfo.drawPlayer.bodyFrame;
 				int num = drawinfo.armorAdjust;
 				bodyFrame.X += num;
@@ -1272,7 +_,7 @@
 			_ = value2 + compositeOffset_BackArm;
 			bodyVect += compositeOffset_BackArm;
 			DrawData drawData;
-			if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < 235) {
+			if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < ArmorIDs.Body.Count) {
 				if (!drawinfo.drawPlayer.invis || IsArmorDrawnWhenInvisible(drawinfo.drawPlayer.body)) {
 					Texture2D value3 = TextureAssets.ArmorBodyComposite[drawinfo.drawPlayer.body].Value;
 					drawData = new DrawData(value3, vector, drawinfo.compTorsoFrame, drawinfo.colorArmorBody, bodyRotation, drawinfo.bodyVect, 1f, drawinfo.playerEffect, 0) {
@@ -1405,7 +_,7 @@
 					drawinfo.DrawDataCache.Add(item);
 				}
 			}
-			else if (drawinfo.drawPlayer.head > 0 && drawinfo.drawPlayer.head < 266 && !flag2) {
+			else if (drawinfo.drawPlayer.head > 0 && drawinfo.drawPlayer.head < ArmorIDs.Head.Count && !flag2) {
 				if (!(drawinfo.drawPlayer.invis && flag3)) {
 					if (drawinfo.drawPlayer.head == 13) {
 						int num6 = 0;
@@ -1766,7 +_,7 @@
 		}
 
 		public static void DrawPlayer_25_Shield(ref PlayerDrawSet drawinfo) {
-			if (drawinfo.drawPlayer.shield <= 0 || drawinfo.drawPlayer.shield >= TextureAssets.AccShield.Length)
+			if (drawinfo.drawPlayer.shield <= 0 || drawinfo.drawPlayer.shield >= 10)
 				return;
 
 			Vector2 zero = Vector2.Zero;
@@ -2147,7 +_,7 @@
 			if (drawinfo.usesCompositeTorso) {
 				DrawPlayer_28_ArmOverItemComposite(ref drawinfo);
 			}
-			else if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < 235) {
+			else if (drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < ArmorIDs.Body.Count) {
 				Rectangle bodyFrame = drawinfo.drawPlayer.bodyFrame;
 				int num = drawinfo.armorAdjust;
 				bodyFrame.X += num;
@@ -2223,7 +_,7 @@
 				vector += new Vector2((!drawinfo.playerEffect.HasFlag(SpriteEffects.FlipHorizontally)) ? 1 : (-1), (!drawinfo.playerEffect.HasFlag(SpriteEffects.FlipVertically)) ? 1 : (-1));
 
 			_ = drawinfo.drawPlayer.invis;
-			bool num = drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < 235;
+			bool num = drawinfo.drawPlayer.body > 0 && drawinfo.drawPlayer.body < ArmorIDs.Body.Count;
 			int num2 = drawinfo.compShoulderOverFrontArm ? 1 : 0;
 			int num3 = (!drawinfo.compShoulderOverFrontArm) ? 1 : 0;
 			int num4 = (!drawinfo.compShoulderOverFrontArm) ? 1 : 0;
