--- src/tModLoader/Terraria/Initializers/ChatInitializer.cs
+++ src/tStandalone/Terraria/Initializers/ChatInitializer.cs
@@ -6,9 +_,9 @@
 
 namespace Terraria.Initializers
 {
-	// @todo: if need be, we can add register tag handlers here
 	public static class ChatInitializer
 	{
+
 		public static void Load() {
 			ChatManager.Register<ColorTagHandler>(new string[2] {
 				"c",
@@ -35,12 +_,20 @@
 				"glyph"
 			});
 
+			ChatManager.Commands.ClearCommands();
+
 			ChatManager.Commands.AddCommand<PartyChatCommand>().AddCommand<RollCommand>().AddCommand<EmoteCommand>()
 				.AddCommand<ListPlayersCommand>()
 				.AddCommand<RockPaperScissorsCommand>()
 				.AddCommand<EmojiCommand>()
 				.AddCommand<HelpCommand>()
 				.AddDefaultCommand<SayChatCommand>();
+
+			if (Main.debugMode) {
+				ChatManager.Commands
+					.AddCommand<SpawnItemCommand>()
+					.AddCommand<SpawnNPCCommand>();
+			}
 
 			for (int i = 0; i < 145; i++) {
 				string name = EmoteID.Search.GetName(i);
