--- src/Terraria/Terraria/Item.cs
+++ src/tModLoader/Terraria/Item.cs
@@ -39,8 +_,8 @@
 		public static int[] headType = new int[266];
 		public static int[] bodyType = new int[235];
 		public static int[] legType = new int[218];
-		public static bool[] staff = new bool[5045];
+		public static bool[] staff = new bool[ItemID.Count];
-		public static bool[] claw = new bool[5045];
+		public static bool[] claw = new bool[ItemID.Count];
 		public bool flame;
 		public bool mech;
 		public int noGrabDelay;
@@ -59,6 +_,9 @@
 		public short makeNPC;
 		public bool expertOnly;
 		public bool expert;
+		#region Modded Variables
+		public bool master;
+		#endregion
 		public bool isAShopItem;
 		public short hairDye = -1;
 		public byte paint;
@@ -170,6 +_,9 @@
 
 		public bool FitsAccessoryVanitySlot {
 			get {
+				if (Main.terrariaPlus && Main.allAccessoriesInVanitySlots)
+					return accessory && balloonSlot != 1;
+
 				if (!canBePlacedInVanityRegardlessOfConditions && handOnSlot == -1 && handOffSlot == -1 && backSlot == -1 && frontSlot == -1 && shoeSlot == -1 && waistSlot == -1 && wingSlot == -1 && shieldSlot == -1 && neckSlot == -1 && faceSlot == -1)
 					return balloonSlot != -1;
 
@@ -43682,6 +_,33 @@
 					DefaultToMusicBox(85);
 					break;
 			}
+			if (type >= 5045) {
+				ModdedSetDefaults(type);
+			}
+		}
+
+		private void ModdedSetDefaults(int type) {
+			switch (type) {
+				case ItemID.SlimeScepter:
+					width = 40;
+					height = 40;
+					summon = true;
+					master = true;
+					autoReuse = false;
+					maxStack = 1;
+					mana = 40;
+					damage = 75;
+					useStyle = ItemUseStyleID.Swing;
+					UseSound = SoundID.Item119;
+					useAnimation = 60;
+					useTime = 60;
+					noMelee = true;
+					knockBack = 9f;
+					value = Item.sellPrice(gold: 1);
+					break;
+			}
+			if (master)
+				rare = -13;
 		}
 
 		private void DefaultToWhip(int projectileId, int dmg, float kb, float shootspeed, int animationTotalTime = 30) {
@@ -44361,7 +_,7 @@
 				playerIndexTheItemIsReservedFor = Main.myPlayer;
 
 			ResetStats(Type);
-			if (type >= 5045)
+			if (type >= ItemID.Count)
 				type = 0;
 
 			if (type == 0) {
@@ -44540,7 +_,7 @@
 				material = ItemID.Sets.IsAMaterial[type];
 
 			RebuildTooltip();
-			if (type > 0 && type < 5045 && ItemID.Sets.Deprecated[type]) {
+			if (type > 0 && type < ItemID.Count && ItemID.Sets.Deprecated[type]) {
 				netID = 0;
 				type = 0;
 				stack = 0;
@@ -44556,6 +_,7 @@
 			shopSpecialCurrency = -1;
 			shopCustomPrice = null;
 			expert = false;
+			master = false;
 			isAShopItem = false;
 			expertOnly = false;
 			instanced = false;
@@ -45840,7 +_,7 @@
 				Prefix(reader.ReadByte());
 			}
 
-			if (type >= 5045)
+			if (type >= ItemID.Count)
 				TurnToAir();
 		}
 
