--- src/Terraria/Terraria/Player.cs
+++ src/tModLoader/Terraria/Player.cs
@@ -498,6 +_,10 @@
 		public float teleportTime;
 		public int teleportStyle;
 		public bool sloping;
+
+		//tStandalone Bools
+		public bool trulyConfused;
+
 		public bool chilled;
 		public bool dazed;
 		public bool frozen;
@@ -686,7 +_,7 @@
 		public const int maxBuffs = 22;
 		public int[] buffType = new int[22];
 		public int[] buffTime = new int[22];
-		public bool[] buffImmune = new bool[323];
+		public bool[] buffImmune = new bool[BuffID.Count];
 		public int heldProj = -1;
 		public int breathCD;
 		public int breathMax = 200;
@@ -1267,7 +_,7 @@
 		public int cUnicornHorn;
 		public int cMinion;
 		public int cLeinShampoo;
-		public int[] ownedProjectileCounts = new int[950];
+		public int[] ownedProjectileCounts = new int[ProjectileID.Count];
 		public bool[] npcTypeNoAggro = new bool[663];
 		public int lastPortalColorIndex;
 		public int _portalPhysicsTime;
@@ -6191,7 +_,7 @@
 
 		private void UpdateItemDye(bool effectiveEquipmentSlot, bool armorHidden, Item armorItem, Item dyeItem) {
 			if (!armorItem.IsAir && (!(effectiveEquipmentSlot && armorHidden) || armorItem.wingSlot > 0 || armorItem.type == 934 || armorItem.type == 4341 || armorItem.type == 4563)) {
-				if (armorItem.handOnSlot > 0 && armorItem.handOnSlot < 22)
+				if (armorItem.handOnSlot > 0 && armorItem.handOnSlot < 23)
 					cHandOn = dyeItem.dye;
 
 				if (armorItem.handOffSlot > 0 && armorItem.handOffSlot < 14)
@@ -6203,7 +_,7 @@
 				if (armorItem.frontSlot > 0 && armorItem.frontSlot < 9)
 					cFront = dyeItem.dye;
 
-				if (armorItem.shoeSlot > 0 && armorItem.shoeSlot < 25)
+				if (armorItem.shoeSlot > 0 && armorItem.shoeSlot < 26)
 					cShoe = dyeItem.dye;
 
 				if (armorItem.waistSlot > 0 && armorItem.waistSlot < 17)
@@ -6273,6 +_,15 @@
 					fireWalk = true;
 					buffImmune[24] = true;
 				}
+
+				//tStandalone Buffs
+				#region tStandalone
+				else if (buffType[j] == BuffID.TrueConfusion) {
+					trulyConfused = true;
+					confused = true;
+				}
+				#endregion tStandalone
+
 				else if (BuffID.Sets.BasicMountData[buffType[j]] != null) {
 					BuffID.Sets.BuffMountData buffMountData = BuffID.Sets.BasicMountData[buffType[j]];
 					mount.SetMount(buffMountData.mountID, this, buffMountData.faceLeft);
@@ -7786,6 +_,15 @@
 				luckNeedsSync = true;
 				oldLuckPotion = luckPotion;
 			}
+
+			//PostUpdateBuffs();
+		}
+
+		private void PostUpdateBuffs() {
+			if (trulyConfused)
+				Filters.Scene.Activate("VerticalMirror");
+			else
+				Filters.Scene.Deactivate("VerticalMirror");
 		}
 
 		private void UpdateProjectileCaches(int i) {
@@ -12327,6 +_,9 @@
 			else
 				extraAccessorySlots = 0;
 
+			//tStandalone Buffs
+			trulyConfused = false;
+
 			fairyBoots = false;
 			moonLordLegs = false;
 			flowerBoots = false;
@@ -15037,6 +_,12 @@
 		}
 
 		public void StatusFromNPC(NPC npc) {
+			//tSTandalone StatusFromNPC
+			#region tStandalone
+			if (Main.masterModeReloaded && Main.masterMode && npc.type == NPCID.BrainofCthulhu)
+				AddBuff(BuffID.TrueConfusion, 120);
+			#endregion tStandalone
+
 			if (Main.expertMode && ((npc.type == 266 && Main.rand.Next(3) == 0) || npc.type == 267)) {
 				int num = Main.rand.Next(9);
 				if (num == 2 || num == 4)
@@ -17379,7 +_,7 @@
 					AddBuff(194, 2, quiet: false);
 			}
 
-			for (int num26 = 0; num26 < 323; num26++) {
+			for (int num26 = 0; num26 < BuffID.Count; num26++) {
 				buffImmune[num26] = false;
 			}
 
@@ -20916,7 +_,7 @@
 				controlUseItem = true;
 			}
 
-			if (flag2 && altFunctionUse == 0 && inventory[selectedItem].shoot > 0 && ProjectileID.Sets.MinionTargettingFeature[inventory[selectedItem].shoot]) {
+			if (flag2 && altFunctionUse == 0 && ((inventory[selectedItem].shoot > 0 && ProjectileID.Sets.MinionTargettingFeature[inventory[selectedItem].shoot]) || inventory[selectedItem].type == ItemID.SlimeScepter)) {
 				altFunctionUse = 1;
 				controlUseItem = true;
 			}
@@ -29082,7 +_,7 @@
 				if (whoAmI == Main.myPlayer && gravDir == 1f && item.mountType != -1 && mount.CanMount(item.mountType, this))
 					mount.SetMount(item.mountType, this);
 
-				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < 950 && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
+				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < ProjectileID.Count && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
 					FreeUpPetsAndMinions(item);
 
 				if (flag3)
@@ -29356,6 +_,8 @@
 					}
 				}
 
+				ModdedUseItem(item);
+
 				if ((item.type == 50 || item.type == 3124 || item.type == 3199) && itemAnimation > 0) {
 					if (Main.rand.Next(2) == 0)
 						Dust.NewDust(base.position, width, height, 15, 0f, 0f, 150, default(Color), 1.1f);
@@ -29402,6 +_,7 @@
 					}
 				}
 
+
 				if (item.type == 4819 && itemAnimation > 0) {
 					Vector2 value2 = Vector2.UnitY.RotatedBy((float)itemAnimation * ((float)Math.PI * 2f) / 30f) * new Vector2(15f, 0f);
 					for (int n = 0; n < 2; n++) {
@@ -29601,6 +_,24 @@
 				JustDroppedAnItem = false;
 		}
 
+		private void ModdedUseItem(Item moddedItem) {
+			if (moddedItem.type == ItemID.SlimeScepter && itemAnimation > 0 && altFunctionUse == 0 && MinionAttackTargetNPC != -1) {
+				if (NPCID.Sets.IsSlime[Main.npc[MinionAttackTargetNPC].type]) {
+					NPC slime = Main.npc[MinionAttackTargetNPC];
+					int explosionID = Projectile.NewProjectile(slime.Center, Vector2.Zero, ProjectileID.SlimeScepterExplosion, moddedItem.damage, moddedItem.knockBack, whoAmI);
+					Projectile explosion = Main.projectile[explosionID];
+					float explosionMultiplier = NPCID.Sets.SlimeExplosionStrength.TryGetValue(slime.type, out float value) ? value : 1f;
+					explosion.width = (int)(explosion.width * explosionMultiplier);
+					explosion.height = (int)(explosion.height * explosionMultiplier);
+					explosion.damage = (int)(explosion.damage * explosionMultiplier);
+					explosion.Center = slime.Center;
+					slime.StrikeNPCNoInteraction(slime.lifeMax, 0f, 0);
+					for (int i = 0; i < 100; i++)
+						Dust.NewDust(explosion.position, explosion.width, explosion.height, 4, Alpha: 150, newColor: new Color(78, 136, 255, 80), Scale: 2f);
+				}
+			}
+		}
+
 		private void ItemCheck_EmitFoodParticles(Item sItem) {
 			if (itemAnimation >= 1) {
 				Color[] array = ItemID.Sets.FoodParticleColors[sItem.type];
@@ -32982,8 +_,24 @@
 				else if (sItem.type == 4956) {
 					int num161 = (itemAnimationMax - itemAnimation) / itemTime;
 					Vector2 vector33 = new Vector2(num2, num3);
-					int num162 = FinalFractalHelper.GetRandomProfileIndex();
-					if (num161 == 0)
+
+					int num162;
+					switch (Main.player[Main.myPlayer].name) {
+						case "Stevie":
+							num162 = FinalFractalHelper.GetRandomProfileIndexFunny();
+							break;
+						case "Scalie":
+						case "Terra":
+							num162 = FinalFractalHelper.GetRandomProfileIndexTerra();
+							break;
+						case "pbone":
+							num162 = FinalFractalHelper.GetRandomProfileIndexCactus();
+							break;
+						default:
+							num162 = FinalFractalHelper.GetRandomProfileIndex();
+							break;
+					}
+					if (num161 == 0 && (Main.player[Main.myPlayer].name != "Stevie" && Main.player[Main.myPlayer].name != "Scalie" && Main.player[Main.myPlayer].name != "Terra" && Main.player[Main.myPlayer].name != "pbone"))
 						num162 = 4956;
 
 					Vector2 value14 = Main.MouseWorld - MountedCenter;
@@ -33348,10 +_,14 @@
 		}
 
 		private void ItemCheck_MinionAltFeatureUse(Item sItem, bool cShoot) {
-			if (sItem.shoot > 0 && ProjectileID.Sets.MinionTargettingFeature[sItem.shoot] && altFunctionUse == 2 && cShoot && ItemTimeIsZero) {
+			if (sItem.shoot > 0 && ProjectileID.Sets.MinionTargettingFeature[sItem.shoot] && cShoot && altFunctionUse == 2 && ItemTimeIsZero) {
 				ApplyItemTime(sItem);
 				MinionNPCTargetAim(doNotDisableIfTheTargetIsTheSame: false);
 			}
+			else if (sItem.type == ItemID.SlimeScepter && cShoot && altFunctionUse == 2 && ItemTimeIsZero) {
+				ApplyItemTime(sItem);
+				TargetSlime(doNotDisableIfTheTargetIsTheSame: false);
+			}
 		}
 
 		private void ItemCheck_TurretAltFeatureUse(Item sItem, bool cShoot) {
@@ -36982,7 +_,7 @@
 								if (num >= 58) {
 									for (int num7 = 0; num7 < 58; num7++) {
 										int num8 = binaryReader.ReadInt32();
-										if (num8 >= 5045) {
+										if (num8 >= ItemID.Count) {
 											player.inventory[num7].netDefaults(0);
 											binaryReader.ReadInt32();
 											binaryReader.ReadByte();
@@ -37001,7 +_,7 @@
 								else {
 									for (int num9 = 0; num9 < 48; num9++) {
 										int num10 = binaryReader.ReadInt32();
-										if (num10 >= 5045) {
+										if (num10 >= ItemID.Count) {
 											player.inventory[num9].netDefaults(0);
 											binaryReader.ReadInt32();
 											binaryReader.ReadByte();
@@ -37019,7 +_,7 @@
 										for (int num11 = 0; num11 < 5; num11++) {
 											if (num11 != 1) {
 												int num12 = binaryReader.ReadInt32();
-												if (num12 >= 5045) {
+												if (num12 >= ItemID.Count) {
 													player.miscEquips[num11].netDefaults(0);
 													binaryReader.ReadByte();
 												}
@@ -37029,7 +_,7 @@
 												}
 
 												num12 = binaryReader.ReadInt32();
-												if (num12 >= 5045) {
+												if (num12 >= ItemID.Count) {
 													player.miscDyes[num11].netDefaults(0);
 													binaryReader.ReadByte();
 												}
@@ -37043,7 +_,7 @@
 									else {
 										for (int num13 = 0; num13 < 5; num13++) {
 											int num14 = binaryReader.ReadInt32();
-											if (num14 >= 5045) {
+											if (num14 >= ItemID.Count) {
 												player.miscEquips[num13].netDefaults(0);
 												binaryReader.ReadByte();
 											}
@@ -37053,7 +_,7 @@
 											}
 
 											num14 = binaryReader.ReadInt32();
-											if (num14 >= 5045) {
+											if (num14 >= ItemID.Count) {
 												player.miscDyes[num13].netDefaults(0);
 												binaryReader.ReadByte();
 											}
@@ -38273,6 +_,20 @@
 			int num = -1;
 			for (int i = 0; i < 200; i++) {
 				if (Main.npc[i].CanBeChasedBy(this) && (num == -1 || Main.npc[i].Hitbox.Distance(mouseWorld) < Main.npc[num].Hitbox.Distance(mouseWorld)))
+					num = i;
+			}
+
+			if (MinionAttackTargetNPC == num && !doNotDisableIfTheTargetIsTheSame)
+				MinionAttackTargetNPC = -1;
+			else
+				MinionAttackTargetNPC = num;
+		}
+
+		public void TargetSlime(bool doNotDisableIfTheTargetIsTheSame) {
+			Vector2 mouseWorld = Main.MouseWorld;
+			int num = -1;
+			for (int i = 0; i < 200; i++) {
+				if (NPCID.Sets.IsSlime[Main.npc[i].type] && Main.npc[i].CanBeChasedBy(this) && (num == -1 || Main.npc[i].Hitbox.Distance(mouseWorld) < Main.npc[num].Hitbox.Distance(mouseWorld)))
 					num = i;
 			}
 
