--- src/Terraria/Terraria/Player.cs
+++ src/tModLoader/Terraria/Player.cs
@@ -686,7 +_,7 @@
 		public const int maxBuffs = 22;
 		public int[] buffType = new int[22];
 		public int[] buffTime = new int[22];
-		public bool[] buffImmune = new bool[323];
+		public bool[] buffImmune = new bool[BuffID.Count];
 		public int heldProj = -1;
 		public int breathCD;
 		public int breathMax = 200;
@@ -1267,7 +_,7 @@
 		public int cUnicornHorn;
 		public int cMinion;
 		public int cLeinShampoo;
-		public int[] ownedProjectileCounts = new int[950];
+		public int[] ownedProjectileCounts = new int[ProjectileID.Count];
 		public bool[] npcTypeNoAggro = new bool[663];
 		public int lastPortalColorIndex;
 		public int _portalPhysicsTime;
@@ -17379,7 +_,7 @@
 					AddBuff(194, 2, quiet: false);
 			}
 
-			for (int num26 = 0; num26 < 323; num26++) {
+			for (int num26 = 0; num26 < BuffID.Count; num26++) {
 				buffImmune[num26] = false;
 			}
 
@@ -29082,7 +_,7 @@
 				if (whoAmI == Main.myPlayer && gravDir == 1f && item.mountType != -1 && mount.CanMount(item.mountType, this))
 					mount.SetMount(item.mountType, this);
 
-				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < 950 && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
+				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < ProjectileID.Count && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
 					FreeUpPetsAndMinions(item);
 
 				if (flag3)
