--- src/Terraria/Terraria/Player.cs
+++ src/tModLoader/Terraria/Player.cs
@@ -498,6 +_,9 @@
 		public float teleportTime;
 		public int teleportStyle;
 		public bool sloping;
+		#region Modded Buff Bools
+		public bool trulyConfused;
+		#endregion
 		public bool chilled;
 		public bool dazed;
 		public bool frozen;
@@ -686,7 +_,7 @@
 		public const int maxBuffs = 22;
 		public int[] buffType = new int[22];
 		public int[] buffTime = new int[22];
-		public bool[] buffImmune = new bool[323];
+		public bool[] buffImmune = new bool[BuffID.Count];
 		public int heldProj = -1;
 		public int breathCD;
 		public int breathMax = 200;
@@ -1267,7 +_,7 @@
 		public int cUnicornHorn;
 		public int cMinion;
 		public int cLeinShampoo;
-		public int[] ownedProjectileCounts = new int[950];
+		public int[] ownedProjectileCounts = new int[ProjectileID.Count];
 		public bool[] npcTypeNoAggro = new bool[663];
 		public int lastPortalColorIndex;
 		public int _portalPhysicsTime;
@@ -6273,6 +_,12 @@
 					fireWalk = true;
 					buffImmune[24] = true;
 				}
+				#region Modded Buffs
+				else if (buffType[j] == BuffID.TrueConfusion) {
+					trulyConfused = true;
+					confused = true;
+				}
+				#endregion
 				else if (BuffID.Sets.BasicMountData[buffType[j]] != null) {
 					BuffID.Sets.BuffMountData buffMountData = BuffID.Sets.BasicMountData[buffType[j]];
 					mount.SetMount(buffMountData.mountID, this, buffMountData.faceLeft);
@@ -7786,6 +_,14 @@
 				luckNeedsSync = true;
 				oldLuckPotion = luckPotion;
 			}
+			PostUpdateBuffs();
+		}
+
+		private void PostUpdateBuffs() {
+			if (trulyConfused)
+				Filters.Scene.Activate("VerticalMirror");
+			else
+				Filters.Scene.Deactivate("VerticalMirror");
 		}
 
 		private void UpdateProjectileCaches(int i) {
@@ -12327,6 +_,9 @@
 			else
 				extraAccessorySlots = 0;
 
+			#region Modded Buff Bools
+			trulyConfused = false;
+			#endregion
 			fairyBoots = false;
 			moonLordLegs = false;
 			flowerBoots = false;
@@ -15037,6 +_,8 @@
 		}
 
 		public void StatusFromNPC(NPC npc) {
+			if (Main.masterModeReloaded && Main.masterMode && npc.type == NPCID.BrainofCthulhu)
+				AddBuff(BuffID.TrueConfusion, 120);
 			if (Main.expertMode && ((npc.type == 266 && Main.rand.Next(3) == 0) || npc.type == 267)) {
 				int num = Main.rand.Next(9);
 				if (num == 2 || num == 4)
@@ -17379,7 +_,7 @@
 					AddBuff(194, 2, quiet: false);
 			}
 
-			for (int num26 = 0; num26 < 323; num26++) {
+			for (int num26 = 0; num26 < BuffID.Count; num26++) {
 				buffImmune[num26] = false;
 			}
 
@@ -29082,7 +_,7 @@
 				if (whoAmI == Main.myPlayer && gravDir == 1f && item.mountType != -1 && mount.CanMount(item.mountType, this))
 					mount.SetMount(item.mountType, this);
 
-				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < 950 && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
+				if ((item.shoot <= 0 || !ProjectileID.Sets.MinionTargettingFeature[item.shoot] || altFunctionUse != 2) && flag3 && whoAmI == Main.myPlayer && item.shoot >= 0 && item.shoot < ProjectileID.Count && (ProjectileID.Sets.LightPet[item.shoot] || Main.projPet[item.shoot]))
 					FreeUpPetsAndMinions(item);
 
 				if (flag3)
@@ -32982,8 +_,23 @@
 				else if (sItem.type == 4956) {
 					int num161 = (itemAnimationMax - itemAnimation) / itemTime;
 					Vector2 vector33 = new Vector2(num2, num3);
-					int num162 = FinalFractalHelper.GetRandomProfileIndex();
-					if (num161 == 0)
+					int num162;
+					switch (Main.player[Main.myPlayer].name) {
+						case "Stevie":
+							num162 = FinalFractalHelper.GetRandomProfileIndexFunny();
+							break;
+						case "Scalie":
+						case "Terra":
+							num162 = FinalFractalHelper.GetRandomProfileIndexTerra();
+							break;
+						case "pbone":
+							num162 = FinalFractalHelper.GetRandomProfileIndexCactus();
+							break;
+						default:
+							num162 = FinalFractalHelper.GetRandomProfileIndex();
+							break;
+					}
+					if (num161 == 0 && (Main.player[Main.myPlayer].name != "Stevie" && Main.player[Main.myPlayer].name != "Scalie" && Main.player[Main.myPlayer].name != "Terra" && Main.player[Main.myPlayer].name != "pbone"))
 						num162 = 4956;
 
 					Vector2 value14 = Main.MouseWorld - MountedCenter;
