--- src/Terraria/Terraria/Terraria.csproj
+++ src/tModLoader/Terraria/Terraria.csproj
@@ -83,13 +_,14 @@
         <EditBinOpts>$(EditBinOpts) /stack:0x800000</EditBinOpts>
     </PropertyGroup>
     <Message Text="Applying editbin $(EditBinOpts)" Importance="high" />
-    <Exec Command="&quot;$(SolutionDir)largeaddresstool\editbin&quot; $(EditBinOpts) &quot;$(TargetPath)&quot;" />
 	<Message Text="Checking headers with dumpbin." Importance="high" />
-	<Exec Command="&quot;$(SolutionDir)largeaddresstool\dumpbin&quot; $(DumpBin) &quot;$(TargetPath)&quot; | findstr &quot;large&quot;" />
   </Target>
   <Target Name="CopyToStreamDir" AfterTargets="EditBin" Condition="$(Configuration.Contains('Windows'))">
     <Message Text="Copying $(OutputName) to Steam Dir..." Importance="high" />
     <Copy SourceFiles="$(TargetPath)" DestinationFiles="$(TerrariaSteamPath)\$(OutputName).exe" />
     <Copy SourceFiles="$(TargetDir)$(OutputName).pdb" DestinationFolder="$(TerrariaSteamPath)" />
+  </Target>
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent">
+    <Exec Command="call &quot;$(DevEnvDir)..\tools\vsdevcmd.bat&quot;&#xD;&#xA;editbin /largeaddressaware &quot;$(TargetPath)&quot;" />
   </Target>
 </Project>
