--- src/Terraria/Terraria/Terraria.csproj
+++ src/tModLoader/Terraria/Terraria.csproj
@@ -24,7 +_,9 @@
     <Reference Include="FNA" Condition="$(DefineConstants.Contains('FNA'))">
       <HintPath>..\..\..\references\FNA.dll</HintPath>
     </Reference>
-    <Reference Include="Ionic.Zip.CF" />
+    <Reference Include="Ionic.Zip.Reduced">
+      <HintPath>..\..\..\references\Ionic.Zip.Reduced.dll</HintPath>
+    </Reference>
     <Reference Include="Microsoft.Xna.Framework" Condition="$(DefineConstants.Contains('XNA'))">
       <Private>true</Private>
     </Reference>
