--- src/Terraria/Terraria/GameContent/TextureAssets.cs
+++ src/tModLoader/Terraria/GameContent/TextureAssets.cs
@@ -1,5 +_,6 @@
 using Microsoft.Xna.Framework.Graphics;
 using ReLogic.Content;
+using Terraria.ID;
 
 namespace Terraria.GameContent
 {
@@ -177,11 +_,11 @@
 		public static Asset<Texture2D>[] NpcToggle = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] HbLock = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] blockReplaceIcon = new Asset<Texture2D>[2];
-		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[323];
+		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[Terraria.ID.BuffID.Count];
-		public static Asset<Texture2D>[] Item = new Asset<Texture2D>[5045];
+		public static Asset<Texture2D>[] Item = new Asset<Texture2D>[ItemID.Count];
-		public static Asset<Texture2D>[] ItemFlame = new Asset<Texture2D>[5045];
+		public static Asset<Texture2D>[] ItemFlame = new Asset<Texture2D>[ItemID.Count];
 		public static Asset<Texture2D>[] Npc = new Asset<Texture2D>[663];
-		public static Asset<Texture2D>[] Projectile = new Asset<Texture2D>[950];
+		public static Asset<Texture2D>[] Projectile = new Asset<Texture2D>[Terraria.ID.ProjectileID.Count];
 		public static Asset<Texture2D>[] Gore = new Asset<Texture2D>[1269];
 		public static Asset<Texture2D>[] BackPack = new Asset<Texture2D>[9];
 		public static Asset<Texture2D> Rain;
@@ -249,6 +_,12 @@
 		public static Asset<Texture2D> Logo2;
 		public static Asset<Texture2D> Logo3;
 		public static Asset<Texture2D> Logo4;
+		#region tStandalone
+		public static Asset<Texture2D> MMRLogo;
+		public static Asset<Texture2D> MMRLogo2;
+		public static Asset<Texture2D> MMRLogo3;
+		public static Asset<Texture2D> MMRLogo4;
+		#endregion tStandalone
 		public static Asset<Texture2D> TextBack;
 		public static Asset<Texture2D> Chat;
 		public static Asset<Texture2D> Chat2;
