--- src/tModLoader/Terraria/ID/ArmorIDs.cs
+++ src/tStandalone/Terraria/ID/ArmorIDs.cs
@@ -281,6 +_,8 @@
 			public const int GraduationCapBlack = 264;
 			public const int BadgersHat = 265;
 			public const int Count = 266;
+			public const int VanillaCount = 266;
+			//TODO: Count
 		}
 
 		public class Body
@@ -516,6 +_,8 @@
 			public const int GraduationGownBlack = 233;
 			public const int DeadMansSweater = 234;
 			public const int Count = 235;
+			public const int VanillaCount = 235;
+			//TODO: Count
 		}
 
 		public class Legs
@@ -717,6 +_,8 @@
 			public const int GraduationGownBlackExtension = 216;
 			public const int MoonLordLegs = 217;
 			public const int Count = 218;
+			public const int VanillaCount = 218;
+			//TODO: Count
 		}
 
 		public class HandOn
@@ -752,6 +_,8 @@
 			public const sbyte BersekerGlove = 20;
 			public const sbyte FrogWebbing = 21;
 			public const int Count = 22;
+			public const int VanillaCount = 22;
+			//TODO: Count
 		}
 
 		public class HandOff
@@ -778,6 +_,8 @@
 			public const sbyte BersekerGlove = 12;
 			public const sbyte FrogWebbing = 13;
 			public const int Count = 14;
+			public const int VanillaCount = 14;
+			//TODO: Count
 		}
 
 		public class Back
@@ -812,6 +_,8 @@
 			public const sbyte BunnyTail = 28;
 			public const sbyte HallowedCape = 29;
 			public const int Count = 30;
+			public const int VanillaCount = 30;
+			//TODO: Count
 		}
 
 		public class Front
@@ -831,6 +_,8 @@
 			public const sbyte VampireShirt = 7;
 			public const sbyte HunterCloak = 8;
 			public const int Count = 9;
+			public const int VanillaCount = 9;
+			//TODO: Count
 		}
 
 		public class Shoe
@@ -861,6 +_,8 @@
 			public const sbyte HellfireTreads = 23;
 			public const sbyte TerrasparkBoots = 24;
 			public const int Count = 25;
+			public const int VanillaCount = 25;
+			//TODO: Count
 		}
 
 		public class Waist
@@ -889,6 +_,8 @@
 			public const sbyte SandstorminaBottle = 15;
 			public const sbyte TreasureMagnet = 16;
 			public const int Count = 17;
+			public const int VanillaCount = 17;
+			//TODO: Count
 		}
 
 		public class Wing
@@ -945,6 +_,8 @@
 			public const sbyte LongTrailRainbowWings = 45;
 			public const sbyte CreativeWings = 46;
 			public const int Count = 47;
+			public const int VanillaCount = 47;
+			//TODO: Count
 		}
 
 		public class Shield
@@ -959,6 +_,8 @@
 			public const sbyte Hero = 8;
 			public const sbyte BouncingShield = 9;
 			public const int Count = 10;
+			public const int VanillaCount = 10;
+			//TODO: Count
 		}
 
 		public class Neck
@@ -974,6 +_,8 @@
 			public const sbyte ApprenticeScarf = 9;
 			public const sbyte Stinger = 10;
 			public const int Count = 11;
+			public const int VanillaCount = 11;
+			//TODO: Count
 		}
 
 		public class Face
@@ -1001,6 +_,8 @@
 			[Obsolete("Deprecated.")]
 			public const sbyte UnusedID = 15;
 			public const sbyte Count = 16;
+			public const sbyte VanillaCOunt = 17;
+			//TODO: Count
 		}
 
 		public class Balloon
@@ -1023,6 +_,8 @@
 			public const sbyte BundledPartyBalloons = 16;
 			public const sbyte BalloonAnimal = 17;
 			public const int Count = 18;
+			public const int VanillaCount = 18;
+			//TODO: Count
 		}
 
 		public class RocketBoots
