--- src/tModLoader/Terraria/ID/ContentSamples.cs
+++ src/tStandalone/Terraria/ID/ContentSamples.cs
@@ -588,7 +_,7 @@
 
 			public static void SetCreativeMenuOrder() {
 				List<Item> list = new List<Item>();
-				for (int i = 1; i < 5045; i++) {
+				for (int i = 1; i < ItemID.Count; i++) {
 					Item item = new Item();
 					item.SetDefaults(i);
 					list.Add(item);
@@ -692,14 +_,14 @@
 
 			ModifyNPCIds();
 			ProjectilesByType.Clear();
-			for (int j = 0; j < 950; j++) {
+			for (int j = 0; j < ProjectileID.Count; j++) {
 				Projectile projectile = new Projectile();
 				projectile.SetDefaults(j);
 				ProjectilesByType[j] = projectile;
 			}
 
 			ItemsByType.Clear();
-			for (int k = 0; k < 5045; k++) {
+			for (int k = 0; k < ItemID.Count; k++) {
 				Item item = new Item();
 				item.SetDefaults(k);
 				ItemsByType[k] = item;
