--- src/tModLoader/Terraria/GameInput/PlayerInputProfile.cs
+++ src/tStandalone/Terraria/GameInput/PlayerInputProfile.cs
@@ -3,8 +_,6 @@
 using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using Terraria.ModLoader;
 
 namespace Terraria.GameInput
 {
@@ -187,12 +_,9 @@
 		}
 
 		public void CopyGameplaySettingsFrom(PlayerInputProfile profile, InputMode mode) {
-			string[] keysToCopy = new string[] {
+			string[] keysToCopy = new string[19] {
 				"MouseLeft",
 				"MouseRight",
-				"MouseMiddle",
-				"MouseXButton1",
-				"MouseXButton2",
 				"Up",
 				"Down",
 				"Left",
@@ -245,21 +_,6 @@
 			};
 
 			CopyKeysFrom(profile, mode, keysToCopy);
-		}
-
-		public void CopyModHotkeySettingsFrom(PlayerInputProfile profile, InputMode mode) {
-			foreach (var modHotkey in HotKeyLoader.HotKeys) {
-				InputModes[mode].KeyStatus[modHotkey.uniqueName].Clear();
-				if (!string.IsNullOrEmpty(modHotkey.defaultKey))
-					InputModes[mode].KeyStatus[modHotkey.uniqueName].Add(modHotkey.defaultKey);
-			}
-		}
-
-		public void CopyIndividualModHotkeySettingsFrom(PlayerInputProfile profile, InputMode mode, string uniqueName) {
-			var modHotKey = HotKeyLoader.modHotKeys[uniqueName];
-			InputModes[mode].KeyStatus[uniqueName].Clear();
-			if (!string.IsNullOrEmpty(modHotKey.defaultKey))
-				InputModes[mode].KeyStatus[uniqueName].Add(modHotKey.defaultKey);
 		}
 
 		public void CopyGamepadSettingsFrom(PlayerInputProfile profile, InputMode mode) {
