--- src/Terraria/Terraria/GameContent/TextureAssets.cs
+++ src/tStandalone/Terraria/GameContent/TextureAssets.cs
@@ -1,9 +_,10 @@
 using Microsoft.Xna.Framework.Graphics;
 using ReLogic.Content;
+using Terraria.ID;
 
 namespace Terraria.GameContent
 {
-	public static class TextureAssets
+	public static partial class TextureAssets
 	{
 		public static class RenderTargets
 		{
@@ -68,12 +_,12 @@
 		public static Asset<Texture2D>[] CuteFishronMount = new Asset<Texture2D>[2];
 		public static Asset<Texture2D> MinecartWoodMount;
 		public static Asset<Texture2D>[] Wings = new Asset<Texture2D>[47];
-		public static Asset<Texture2D>[] ArmorHead = new Asset<Texture2D>[266];
+		public static Asset<Texture2D>[] ArmorHead = new Asset<Texture2D>[ArmorIDs.Head.Count];
-		public static Asset<Texture2D>[] ArmorBody = new Asset<Texture2D>[235];
+		public static Asset<Texture2D>[] ArmorBody = new Asset<Texture2D>[ArmorIDs.Body.Count];
-		public static Asset<Texture2D>[] ArmorBodyComposite = new Asset<Texture2D>[235];
+		public static Asset<Texture2D>[] ArmorBodyComposite = new Asset<Texture2D>[ArmorIDs.Body.Count];
-		public static Asset<Texture2D>[] FemaleBody = new Asset<Texture2D>[235];
+		public static Asset<Texture2D>[] FemaleBody = new Asset<Texture2D>[ArmorIDs.Body.Count];
-		public static Asset<Texture2D>[] ArmorArm = new Asset<Texture2D>[235];
+		public static Asset<Texture2D>[] ArmorArm = new Asset<Texture2D>[ArmorIDs.Body.Count];
-		public static Asset<Texture2D>[] ArmorLeg = new Asset<Texture2D>[218];
+		public static Asset<Texture2D>[] ArmorLeg = new Asset<Texture2D>[ArmorIDs.Legs.Count];
 		public static Asset<Texture2D>[] AccHandsOn = new Asset<Texture2D>[22];
 		public static Asset<Texture2D>[] AccHandsOnComposite = new Asset<Texture2D>[22];
 		public static Asset<Texture2D>[] AccHandsOff = new Asset<Texture2D>[14];
@@ -177,15 +_,15 @@
 		public static Asset<Texture2D>[] NpcToggle = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] HbLock = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] blockReplaceIcon = new Asset<Texture2D>[2];
-		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[323];
+		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[BuffID.Count];
-		public static Asset<Texture2D>[] Item = new Asset<Texture2D>[5045];
+		public static Asset<Texture2D>[] Item = new Asset<Texture2D>[ItemID.Count];
-		public static Asset<Texture2D>[] ItemFlame = new Asset<Texture2D>[5045];
+		public static Asset<Texture2D>[] ItemFlame = new Asset<Texture2D>[ItemID.Count];
-		public static Asset<Texture2D>[] Npc = new Asset<Texture2D>[663];
+		public static Asset<Texture2D>[] Npc = new Asset<Texture2D>[NPCID.Count];
 		public static Asset<Texture2D>[] Projectile = new Asset<Texture2D>[950];
 		public static Asset<Texture2D>[] Gore = new Asset<Texture2D>[1269];
 		public static Asset<Texture2D>[] BackPack = new Asset<Texture2D>[9];
 		public static Asset<Texture2D> Rain;
-		public static Asset<Texture2D>[] GlowMask = new Asset<Texture2D>[301];
+		public static Asset<Texture2D>[] GlowMask = new Asset<Texture2D>[GlowMaskID.Count];
 		public static Asset<Texture2D>[] Extra = new Asset<Texture2D>[212];
 		public static Asset<Texture2D>[] HighlightMask = new Asset<Texture2D>[623];
 		public static Asset<Texture2D>[] Coin = new Asset<Texture2D>[4];
