--- src/tModLoader/Terraria/GameContent/TilePaintSystemV2.cs
+++ src/tStandalone/Terraria/GameContent/TilePaintSystemV2.cs
@@ -74,7 +_,7 @@
 			public TreeFoliageVariantKey Key;
 
 			public override void Prepare() {
-				Asset<Texture2D> asset = TextureAssets.TreeTop[Key.TextureIndex];
+				Asset<Texture2D> asset = Main.Assets.Request<Texture2D>(TextureAssets.TreeTop[Key.TextureIndex].Name);
 				PrepareTextureIfNecessary(asset.Value);
 			}
 
@@ -88,7 +_,7 @@
 			public TreeFoliageVariantKey Key;
 
 			public override void Prepare() {
-				Asset<Texture2D> asset = TextureAssets.TreeBranch[Key.TextureIndex];
+				Asset<Texture2D> asset = Main.Assets.Request<Texture2D>(TextureAssets.TreeBranch[Key.TextureIndex].Name);
 				PrepareTextureIfNecessary(asset.Value);
 			}
 
@@ -102,7 +_,7 @@
 			public TileVariationkey Key;
 
 			public override void Prepare() {
-				Asset<Texture2D> asset = TextureAssets.Tile[Key.TileType];
+				Asset<Texture2D> asset = Main.Assets.Request<Texture2D>(TextureAssets.Tile[Key.TileType].Name);
 				PrepareTextureIfNecessary(asset.Value);
 			}
 
@@ -116,7 +_,7 @@
 			public WallVariationKey Key;
 
 			public override void Prepare() {
-				Asset<Texture2D> asset = TextureAssets.Wall[Key.WallType];
+				Asset<Texture2D> asset = Main.Assets.Request<Texture2D>(TextureAssets.Wall[Key.WallType].Name);
 				PrepareTextureIfNecessary(asset.Value);
 			}
 
