--- src/Terraria/Terraria/GameContent/TownRoomManager.cs
+++ src/tStandalone/Terraria/GameContent/TownRoomManager.cs
@@ -9,7 +_,7 @@
 	public class TownRoomManager
 	{
 		private List<Tuple<int, Point>> _roomLocationPairs = new List<Tuple<int, Point>>();
-		private bool[] _hasRoom = new bool[663];
+		private bool[] _hasRoom = new bool[NPCID.Count];
 
 		public void AddOccupantsToList(int x, int y, List<int> occupantsList) {
 			AddOccupantsToList(new Point(x, y), occupantsList);
