--- src/tModLoader/Terraria/Map/MapHelper.cs
+++ src/tStandalone/Terraria/Map/MapHelper.cs
@@ -5,8 +_,6 @@
 using System.IO;
 using System.Threading;
 using Terraria.IO;
-using Terraria.ModLoader;
-using Terraria.ModLoader.IO;
 using Terraria.Social;
 using Terraria.Utilities;
 
@@ -112,8 +_,7 @@
 		private static ushort dirtPosition;
 		private static ushort rockPosition;
 		private static ushort hellPosition;
-		internal static ushort modPosition;
-		internal static Color[] colorLookup;
+		private static Color[] colorLookup;
 		private static ushort[] snowTypes;
 		private static ushort wallRangeStart;
 		private static ushort wallRangeEnd;
@@ -1439,7 +_,6 @@
 
 			hellPosition = num12;
 			colorLookup[num12] = color9;
-			modPosition = (ushort)(num12 + 1);
 			snowTypes = new ushort[6];
 			snowTypes[0] = tileLookup[147];
 			snowTypes[1] = tileLookup[161];
@@ -1629,9 +_,7 @@
 				}
 			}
 
-			ushort mapType = (ushort)(num3 + baseOption);
-			MapLoader.ModMapOption(ref mapType, i, j);
-			return MapTile.Create(mapType, (byte)num2, (byte)num);
+			return MapTile.Create((ushort)(num3 + baseOption), (byte)num2, (byte)num);
 		}
 
 		public static void GetTileBaseOption(int y, Tile tileCache, ref int baseOption) {
@@ -2244,7 +_,7 @@
 								byte b5 = 0;
 								int num7;
 								ushort num8;
-								if (mapTile.Light <= 18 || mapTile.Type >= modPosition) {
+								if (mapTile.Light <= 18) {
 									flag2 = false;
 									flag = false;
 									num7 = 0;
@@ -2388,9 +_,7 @@
 
 						deflateStream.Dispose();
 						FileUtilities.WriteAllBytes(text, memoryStream.ToArray(), isCloudSave);
-						//patch file: text
 					}
-					MapIO.WriteModFile(text, isCloudSave);
 				}
 			}
 
