--- src/tModLoader/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
+++ src/tStandalone/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
@@ -99,7 +_,7 @@
 			AchievementCategory category = _achievement.Category;
 			vector.Y += 2f;
 			vector.X += 4f;
-			spriteBatch.Draw(_categoryTexture.Value, vector, _categoryTexture.Frame(4, 2, (int)category), base.IsMouseHovering ? Color.White : Color.Silver, 0f, Vector2.Zero, 0.5f, SpriteEffects.None, 0f);
+			spriteBatch.Draw(_categoryTexture.Value, vector, _categoryTexture.Frame(5, 2, (int)category), base.IsMouseHovering ? Color.White : Color.Silver, 0f, Vector2.Zero, 0.5f, SpriteEffects.None, 0f);
 			vector.X += 4f;
 			vector.X += 17f;
 			ChatManager.DrawColorCodedStringWithShadow(spriteBatch, FontAssets.ItemStack.Value, _achievement.FriendlyName.Value, vector, value3, 0f, Vector2.Zero, baseScale, num3);
