--- src/Terraria/Terraria/GameContent/UI/States/UIAchievementsMenu.cs
+++ src/tModLoader/Terraria/GameContent/UI/States/UIAchievementsMenu.cs
@@ -85,7 +_,7 @@
 			uIElement2.Height.Set(32f, 0f);
 			uIElement2.Top.Set(10f, 0f);
 			Asset<Texture2D> texture = Main.Assets.Request<Texture2D>("Images/UI/Achievement_Categories");
-			for (int j = 0; j < 4; j++) {
+			for (int j = 0; j < 5; j++) {
 				UIToggleImage uIToggleImage = new UIToggleImage(texture, 32, 32, new Point(34 * j, 0), new Point(34 * j, 34));
 				uIToggleImage.Left.Set(j * 36 + 8, 0f);
 				uIToggleImage.SetState(value: true);
@@ -103,14 +_,17 @@
 				if (_categoryButtons[i].IsMouseHovering) {
 					string text = "";
 					switch (i) {
+						case 4:
+							text = Language.GetTextValue("Modded.ModdedCategory");
+							break;
 						case 3:
 							text = Language.GetTextValue("Achievements.ChallengerCategory");
 							break;
-						case 1:
-							text = Language.GetTextValue("Achievements.CollectorCategory");
-							break;
 						case 2:
 							text = Language.GetTextValue("Achievements.ExplorerCategory");
+							break;
+						case 1:
+							text = Language.GetTextValue("Achievements.CollectorCategory");
 							break;
 						case 0:
 							text = Language.GetTextValue("Achievements.SlayerCategory");
