--- src/Terraria/Terraria/NetMessage.cs
+++ src/tStandalone/Terraria/NetMessage.cs
@@ -15,6 +_,7 @@
 
 namespace Terraria
 {
+	// TODO: moddedAIArray :shrug:
 	public class NetMessage
 	{
 		public struct NetSoundInfo
@@ -89,7 +_,7 @@
 				writer.Write((byte)msgType);
 				switch (msgType) {
 					case 1:
-						writer.Write("Terraria" + 230);
+						writer.Write("tStandalone");
 						break;
 					case 2:
 						text.Serialize(writer);
@@ -579,7 +_,7 @@
 								}
 							}
 
-							if (nPC2.type >= 0 && nPC2.type < 663 && Main.npcCatchable[nPC2.type])
+							if (nPC2.type >= 0 && nPC2.type < NPCID.Count && Main.npcCatchable[nPC2.type])
 								writer.Write((byte)nPC2.releaseOwner);
 
 							break;
