--- src/tModLoader/Terraria/PopupText.cs
+++ src/tStandalone/Terraria/PopupText.cs
@@ -1,4 +_,5 @@
 using Microsoft.Xna.Framework;
+using Terraria.Enums;
 using Terraria.GameContent;
 using Terraria.Localization;
 
@@ -184,6 +_,7 @@
 				Main.popupText[num2].position.X = newItem.position.X + (float)newItem.width * 0.5f - vector2.X * 0.5f;
 				Main.popupText[num2].position.Y = newItem.position.Y + (float)newItem.height * 0.25f - vector2.Y * 0.5f;
 				Main.popupText[num2].color = Color.White;
+				Main.popupText[num2].master = false;
 				if (newItem.rare == 1)
 					Main.popupText[num2].color = new Color(150, 150, 255);
 				else if (newItem.rare == 2)
@@ -394,7 +_,7 @@
 
 			if (expert)
 				color = new Color((byte)Main.DiscoR, (byte)Main.DiscoG, (byte)Main.DiscoB, Main.mouseTextColor);
-			else if (master)
+			if (master)
 				color = new Color(255, (byte)(Main.masterColor * 200f), 0, Main.mouseTextColor);
 
 			bool flag = false;
