--- src/tModLoader/Terraria/RecipeGroup.cs
+++ src/tStandalone/Terraria/RecipeGroup.cs
@@ -8,7 +_,6 @@
 		public Func<string> GetText;
 		public HashSet<int> ValidItems;
 		public int IconicItemId;
-		internal bool[] ValidItemsLookup;
 		public static Dictionary<int, RecipeGroup> recipeGroups = new Dictionary<int, RecipeGroup>();
 		public static Dictionary<string, int> recipeGroupIDs = new Dictionary<string, int>();
 		public static int nextRecipeGroupIndex;
@@ -24,13 +_,6 @@
 			recipeGroups.Add(num, rec);
 			recipeGroupIDs.Add(name, num);
 			return num;
-		}
-
-		public bool ContainsItem(int type) {
-			if (ValidItemsLookup == null)
-				return ValidItems.Contains(type);
-
-			return ValidItemsLookup[type];
 		}
 	}
 }
