--- src/tModLoader/Terraria/GameContent/Tile_Entities/TEItemFrame.cs
+++ src/tStandalone/Terraria/GameContent/Tile_Entities/TEItemFrame.cs
@@ -1,6 +_,5 @@
 using System.IO;
 using Terraria.DataStructures;
-using Terraria.ID;
 
 namespace Terraria.GameContent.Tile_Entities
 {
@@ -72,7 +_,7 @@
 		}
 
 		public override void WriteExtraData(BinaryWriter writer, bool networkSend) {
-			writer.Write((short)(item.netID >= ItemID.Count ? 0 : item.netID));
+			writer.Write((short)item.netID);
 			writer.Write(item.prefix);
 			writer.Write((short)item.stack);
 		}
