--- src/Terraria/Terraria/Initializers/ChatInitializer.cs
+++ src/tStandalone/Terraria/Initializers/ChatInitializer.cs
@@ -41,6 +_,10 @@
 				.AddCommand<HelpCommand>()
 				.AddDefaultCommand<SayChatCommand>();
 
+#if DEBUG
+			ChatManager.Commands.AddCommand<SpawnItemCommand>().AddCommand<SpawnNPCCommand>(); // Added by tStandalone.
+#endif
+
 			for (int i = 0; i < 145; i++) {
 				string name = EmoteID.Search.GetName(i);
 				string key = "EmojiCommand." + name;
