--- src/tModLoader/Terraria/Initializers/ScreenEffectInitializer.cs
+++ src/tStandalone/Terraria/Initializers/ScreenEffectInitializer.cs
@@ -9,6 +_,9 @@
 	public static class ScreenEffectInitializer
 	{
 		public static void Load() {
+			#region Modded Screen Shaders
+			Filters.Scene["VerticalMirror"] = new Filter(new ScreenShaderData(Main.VerticalMirrorRef, "VerticallyMirror"), EffectPriority.Medium);
+			#endregion
 			Filters.Scene["Nebula"] = new Filter(new ScreenShaderData("FilterTower").UseColor(1f, 0f, 0.9f).UseOpacity(0.35f), EffectPriority.High);
 			Filters.Scene["Solar"] = new Filter(new ScreenShaderData("FilterTower").UseColor(1f, 0.7f, 0f).UseOpacity(0.3f), EffectPriority.High);
 			Filters.Scene["Stardust"] = new Filter(new ScreenShaderData("FilterTower").UseColor(0f, 0.5f, 1f).UseOpacity(0.5f), EffectPriority.High);
